src/solver/conv_hip_implicit_gemm_mlir_bin_fwd.cpp

```diff
@@ -29,6 +29,11 @@
 #include <miopen/solver/implicitgemm_util.hpp>
 #include <miopen/handle.hpp>
 
+#ifdef __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wunused-function"
+#endif
```

Comment:
Why you need to suppress this? Bug in clang-tidy?

Reply:
Perhaps `static` instead of unnamed namespace can be used as a workaround?

Reply:
I don't blame clang-tidy for this. It's basically because `CalculateGemmSize()` is only used in maro-ed section of the code in `IsApplicable()`. Therefore, tidy target cannot see its call-site, and complained that there's a function being defined but never used.

Reply:
In other words, we don't build `clang-tidy` or `opencl-tidy` target with `-MLIR_USE_MIOPEN` on caused it. But this is all intended.

Reply:
Aha. Just enclose the function into #if... #endif

Reply:
And please remove suppression.

Reply:
Done in 57a7189.